providers/implementations/exchange/dh_exch.c

```diff
@@ -104,7 +104,7 @@ static int dh_init(void *vpdhctx, void *vdh)
     DH_free(pdhctx->dh);
     pdhctx->dh = vdh;
     pdhctx->kdf_type = PROV_DH_KDF_NONE;
-    return ossl_dh_check_key(vdh);
+    return ossl_dh_check_key(vdh) && dh_set_ctx_params(pdhctx, params);
```

Comment:
I dont know that is the right order..
Should it set the params and then do the check_key() ??

Reply:
Happy to swap them around.  I wasn't sure.